### DISCUSSION TITLE
For the osu!mania added does not show hitsound diff switch

### DISCUSSION THREAD

**Discussion by mas-alone** [ppy/osu#13595] [Ideas] [answered]
osu!mania game mode beatmap usually requires a hitsound diff, which is not required to be played by the player, but is generally used to give suggestions to modders. Now the default is to display all diff, we hope to add a switch in the Settings or song selection interface. Used to select whether to show or not show the hitsound diff

It's like this beatmap
https://osu.ppy.sh/beatmapsets/1417233#mania/2920287

has a 7k hitsound diff. but is 4k map

**Comment by peppy** [comment, marked as answer]
I have no idea what the point of this is. Please link to documentation on why this is required so we can understand the root issue rather than implement weird hacks/workarounds.

**Reply by mas-alone** [reply]
Hitsound diff is osu!mania game mode Ranking Criteria is required, but now the hitsound diff is displayed in the game as well, which will cause some players to incorrectly open it and play. This is also true in the stable version. It's not a big deal, I just hope it's better to hide it and I'll post a picture after work tonight to explain it.
![IMG_20210621_105229](https://user-images.githubusercontent.com/52590027/122701033-f69b0980-d27e-11eb-8ce9-4b91fcbd93c3.jpg)

**Reply by gagahpangeran** [reply]
From what I know, the hit sound only difficulty is required when you want to rank the osu!mania beatmap so BN can easily check the hit sound part only. That difficulty will be removed after the check and before the map gets rank.

**Reply by mas-alone** [reply]
Yes, but some players will still click on the HitSound diff to play, and after the difficulty is removed, it will remain in the player's local client. I think mod and play should be separated, right? Add a switch to hide it. Bn needs to check hitSound by just clicking the show button

**Reply by Joehuu** [reply]
In lazer you can hide, but it'll remain in your disk. Maybe once beatmap updating becomes a thing, we can delete individual unsubmitted difficulties?

<img width="274" alt="Screen Shot 2021-06-20 at 8 29 01 PM" src="https://user-images.githubusercontent.com/35318437/122703334-29ed8200-d206-11eb-8826-8d51378e9425.png">

**Reply by mas-alone** [reply]
Manual hiding is also a good idea, but I think auto-hiding is smarter, because hitSound diff is only useful when you modify the beatmap(for Ranked), so it's useless for normal play. Players should be more focused on playing when opening a song list, right

**Reply by mas-alone** [reply]
It would be nice if Lazer could automatically update the beatmap synchronously, and automatically delete the local data when a diff is deleted

**Reply by peppy** [reply]
This is just stupid. The ranking criteria needs to be fixed, not the game.

If it's an editor shortcoming(?) then the editor should be improved resulting in the ranking criteria getting fixed.

We are not hacking around issues with lazer.

**Reply by peppy** [reply]
In checking with the team, this is not written anywhere in the ranking criteria and for the time being, should be up to you as a user to hide or delete the difficulties if you don't have a use for them any more. They are always removed before ranking so this should not affect the larger audience.

We will likely add functionality in the editor in the not-so-near future to provide alternatives to this particular process, but this is too far off to make any calls on just yet.

**Reply by mas-alone** [reply]
> In checking with the team, this is not written anywhere in the ranking criteria and for the time being, should be up to you as a user to hide or delete the difficulties if you don't have a use for them any more. They are always removed before ranking so this should not affect the larger audience.
> 
> We will likely add functionality in the editor in the not-so-near future to provide alternatives to this particular process, but this is too far off to make any calls on just yet.

sorry my eng not good. This is also a subjective suggestion. This looks great. Thank you for your efforts.

**Reply by Nivalyx** [reply]
Ah, I was made aware of this discussion thread earlier and I'll be posting this here as well for clarity/to avoid further confusion I guess :

@mas-alone I'm fairly certain that you're misunderstanding something in assuming that "`[Hitsound]` diff is required by the Ranking Criteria". There's no such rule in the Ranking Criteria, and there has never been -- although the Ranking Criteria mentions that "map have to be hitsounded", the Ranking Criteria never explicitly tells mappers *how* they should hitsound their map if any.

While it's true that the practice of mappers creating a temporary "hitsound diffs" to help them hitsound the mapset they're working on is becoming more and more common nowadays, you're free to hitsound your map in any way you want! You can, for example, hitsound all of the objects you have across a single mapset manually and you'll still be able to rank the said mapset without problem (provided that nothing's broken or out of place, ofc). 

It's completely up to you really -- all in all, having a `[Hitsound]` diff is *not* a must.

---

Also for the record "dummy" difficulties like these (debug diffs, hitsound diffs, etc.) are somewhat to be expected on maps that are Pending/WIP so yeah there's that 🐇
